platform/server/src/KafkaSubEngine.ts

```diff
@@ -0,0 +1,154 @@
+import * as Kafka from 'node-rdkafka';
+import { PubSubEngine } from 'graphql-subscriptions';
+import { EventEmitter } from 'events';
+import { ObjectID } from 'mongodb';
+
+export interface IKafkaOptions {
+  topic: string
+  host: string
+  port: string
+  modelName: string
+}
+
+export class KafkaSubEngine extends PubSubEngine {
```

Comment:
What is the reason for using a custom engine over https://github.com/ancashoria/graphql-kafka-subscriptions?

The reason I am asking is I am looking to add a Kafka template to Graphback, do we need a custom one?

Reply:
@craicoverflow good question:

- This needed to be able to parse Debezium published change events into a structure that Graphback can understand. https://github.com/aerogear/OpenVolunteerPlatform/blob/a4c9492238888c73637429c879243ef9b2a5bc20/platform/server/src/KafkaSubEngine.ts#L105-L135

You can go with the library later on if changes will be handled by the application. But for this, the application has to handle topic creation too.

Reply:
Thanks for the info